docs/fastdds/dds_layer/topic/instances.rst

```diff
@@ -89,18 +101,111 @@ The following IDL may define the data model of this specific example:
+    :dedent: 12
+    :start-after: //WRITE-NON-REGISTERED-INSTANCE
+    :end-before: //!
+
+.. warning::
+
+  Managing correctly the instance handles in the user application is paramount.
+  Otherwise, a sample corresponding to a different instance could wrongly update the instance which handle the user has
+  passed to the operation (if a non ``NIL`` instance is provided, the instance handle is not recalculated, trusting
+  that the one passed by the user is the correct one).
+  The following code updates the first instance of this example with the information coming from the second instance.
+
+  .. literalinclude:: /../code/DDSCodeTester.cpp
+      :language: c++
+      :dedent: 12
+      :start-after: //WRONG-INSTANCE-UPDATE
+      :end-before: //!
+
+Once the plane leave the controlled area, the air traffic control center would unregister the instance.
+Unregistering implies that the DataWriter for this specific center has no more information about the unregistered
+instance, and in this way the matched DataReaders are notified.
+The flight is still in the air but out of scope of this particular DataWriter. The instance is alive but no longer
+tracked by this center.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+    :language: c++
+    :dedent: 12
+    :start-after: //UNREGISTER-INSTANCE
+    :end-before: //!
+
+Finally, when the flight lands the instance is disposed.
+This means in this specific example that as far as the DataWriter knows, the instance no longer exists and should be
+considered not alive.
+With this operation, the DataWriter conveys this information to the matched DataReaders.
+
+.. literalinclude:: /../code/DDSCodeTester.cpp
+    :language: c++
+    :dedent: 12
+    :start-after: //DISPOSE-INSTANCE
+    :end-before: //!
+
+From the management center point of view, the samples are read using the same DataReader subscribed to the Topic where
+the instances are being published.
+However, |SampleInfo::valid_data-api| must be checked to ensure that the sample received contains a data sample.
+Otherwise, a change of the instance state is being notified.
+:ref:`dds_layer_topic_instance_lifecycle` contains a diagram showing the instance statechart.
 
 .. literalinclude:: /../code/DDSCodeTester.cpp
-   :language: c++
-   :dedent: 12
-   :start-after: //INSTANCES
-   :end-before: //!
+    :language: c++
+    :dedent: 12
+    :start-after: //READING-INSTANCE
+    :end-before: //!
 
 Relational databases
 ^^^^^^^^^^^^^^^^^^^^
 
-Instances could also be used as a relational database.
-The instance key is analogous to the primary key in the database (unique identifier of something within the data).
-In this scenario, writing a new sample to the instance is similar to insert/update the database; and disposing the
-instance is like deleting the data corresponding to the given primary key from the database.
+Consider now that the air traffic management center wants to keep a database with the flights being tracked.
```

Comment:
In general, I'd elaborate a bit more in how each of the operations (register instance, write, unregister, and dispose) would mean in each of the two DB cases